01_fib/Jake88/main.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fib(7)
+}
+
+func fib(n int) {
+	// Our initial map, with the default starting values needed to begin the sequence.
+	fibonacciMap := map[int]int{
```

Comment:
This map is overkill. You only use two of its value at any given time, and it is never used once the function returns.
I would suggest you store two variables (perhaps `current` and `previous`) initialized to `1` and `0`, then update them in the loop accordingly.
Alternatively, you can store the map as an unexported `var` outside this function to turn it into a global variable,
```
var fibonacciMap = map[int]int{
  -1: 1,
  0: 0,
}
```
If you go down this path, you will find it harder to write the fibonacci function (and may find it easier to write a recursive function that first checks the map, then populates it if it doesn't exist before returning).
I wouldn't recommend this though.

Reply:
Actioned. Converted to two variables instead.